### ISSUE TITLE
Pitch (Y-Axis) Rotation is Backwards

### ISSUE THREAD

**Issue by SterlingPeet** [arindam1993/MultiRobotEnvironment#5]
In left-handed coordinates (much like right-handed coordinates) the positive rotation around an axis is defined such that when you align your thumb with the positive axis direction, your fingers curl around the axis in the positive rotation direction:

![aerobics](https://cloud.githubusercontent.com/assets/1963350/6632191/94a1270e-c908-11e4-9b40-14fafa24a984.png)

(image from the POVray documentation)

Careful inspection and setting up several tests has yielded an error:

![pitchrotbackwards](https://cloud.githubusercontent.com/assets/1963350/6632206/bdd6fe5a-c908-11e4-9a2d-46f7112d31df.gif)

This test robot commanded positive pitch for 4 seconds.  Clearly, it does not follow the definition illustrated above.

Simple tests for translation in X, Y and Z, as well as rotation for Yaw and Roll behave appropriately for a left-handed coordinate system.

**Comment by SterlingPeet** [comment]
Ooops, almost forgot, the specific file in question is the `LinearAlgebraUtils.py` file.

Additionally, if you incorporate the cached-trig speedup from piazza, the original implementation posted there has the "flaw" of being a right-handed rotation matrix.

**Comment by arindam1993** [comment]
Thanks Sterling, this helps a ton!
I believe I found the issue, in `LinearAlegerbraUtils.py`, `getRotMatrixFromYPR()` the sign of b has a -ve sign because of a typo(I suck).
The reason I never ran into an issue was because I implemented  `getYPRFromVector()` to achieve  Look At Position functionality,(I use the result to get a rotation required for the agent to face the ball, and i return this as deltaRot),  with an equivalent sign error on calculating the pitch nullifying the two errors out, and I always used this function to calculate rotations.
My guess is you implemented you own Look At function with no sign errors and thus ran into a bunch of problems.
Fixed it just now, will post results shortly.
Thanks a ton, once again, you took a bunch of effort on this.

EDIT: I think the handedness the matrix is fine, I do take the transpose, which is probably what slows it down.

RESULTS: Yup my hunch was right, thanks a ton Sterling pushed a fix, whew.
